staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go

```diff
@@ -131,16 +131,21 @@ func (s *Server) Start(ctx context.Context) error {
 	}
 
 	var shutdownCh <-chan struct{}
+	var listenerStoppedCh <-chan struct{}
 	if s.secureServing != nil {
 		var err error
-		shutdownCh, err = s.secureServing.Serve(mux, 0, ctx.Done())
+		shutdownCh, listenerStoppedCh, err = s.secureServing.Serve(mux, 0, ctx.Done())
 		if err != nil {
 			return fmt.Errorf("failed to start secure server: %w", err)
 		}
 	}
 
+	<-listenerStoppedCh
+	klog.V(1).InfoS("[graceful-termination] HTTP Server has stopped listening")
```

Comment:
is this the right order? the listener stops before all the requests have completed being handled?

Reply:
it's not super clear to me why we would want to block exit after the requests have completed being handled (`<-shutdownCh`)

Reply:
> is this the right order? the listener stops before all the requests have completed being handled?

Listener != server. So yes, we stop listening for new requests before in-flights requests are all finished.